driver-sync/src/main/com/mongodb/client/internal/CommandMarker.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.internal;
+
+import com.mongodb.Block;
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientException;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoTimeoutException;
+import com.mongodb.ReadConcern;
+import com.mongodb.ReadPreference;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.connection.ClusterSettings;
+import org.bson.RawBsonDocument;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+@SuppressWarnings("UseOfProcessBuilder")
+class CommandMarker implements Closeable {
+    private MongoClient client;
+    private final ProcessBuilder processBuilder;
+    private boolean active;
+
+    CommandMarker(final Map<String, Object> options) {
+        String connectionString;
+
+        if (options.containsKey("mongocryptdURI")) {
+            connectionString = (String) options.get("mongocryptdURI");
+        } else if (System.getProperty("os.name").toLowerCase().contains("windows")) {
+            connectionString = "mongodb://localhost:27020";
+        } else {
+            connectionString = "mongodb://%2Ftmp%2Fmongocryptd.sock";
```

Comment:
This may be a bad idea as our dependency on jnr.unixsocket is optional.  I guess we either have to remove this or make the dependency required.